feat!: removes deprecated v1 certificate behavior

deborahgu opened the pull request:
## Description

this removes the long-deprecated v1  certificate behavior. This removes the old-style date selection behavior  (ie., not a choice between *Immediately upon passing*, *End date of course*, *A date after the course end date*), which is no longer reliably maintained or supported in Studio or Credentials.

There were no special concerns raised either for the deprecation ticket or for the [corresponding discussion](https://discuss.openedx.org/t/deprecation-removal-waffle-flag-enable-v2-cert-display-settings/13806)

FIXES: #35399

deborahgu committed:
feat!: removes deprecated v1 certificate behavior

this removes the long-deprecated v1  certificate behavior. This removes
the old-style date selection behavior  (ie., not a choice between
*Immediately upon passing*, *End date of course*, *A date after the course
end date*), which is no longer reliably maintained or supported in
Studio or Credentials.

FIXES: #35399

deborahgu committed:
chore: linter fixes

Cleanup of some blank line issues

deborahgu committed:
chore: lint fixes

removing unused imports

deborahgu committed:
chore: fixed test

*  fixing a test which was replying on implicit behavior in the choices
   between certificate display behaviors
* lint error

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-3199_removeENABLE_V2_CERT_DISPLAY_SETTINGS

justinhynes commented on lms/envs/common.py:
```diff
@@ -949,17 +949,6 @@
-    # .. settings. Will eventually become the default.
-    # .. toggle_use_cases: temporary
-    # .. toggle_creation_date: 2021-07-26
-    # .. toggle_target_removal_date: 2021-10-01
```
Hey, looks like we nailed the day, just not the year. 😆😭

justinhynes commented:
Assumption: I believe that the Studio frontend has been converted to an MFE -- the [frontend-app-authoring MFE](https://github.com/openedx/frontend-app-authoring).

I assume that the V2 functionality has been carried over, and didn't reimplement the v1 vs. v2 functionality. I wonder if we should create another ticket to take a peek to see if any remnants have made it into the MFE?

justinhynes commented on lms/djangoapps/learner_home/serializers.py:
```diff
@@ -301,21 +301,18 @@ def get_availableDate(self, enrollment):
         course_overview = enrollment.course_overview
         available_date = course_overview.certificate_available_date
```
🤔 Should we change the default of this now that the old way to do things has been removed? Or maybe it should be an `else`/default case as part of the if/elif?

I know we've had issues with holding on to stale data in the past and I'm just trying to think of this from a defensive standpoint. If the display behavior is "immediately upon passing" or the course is self-paced, the cert available date should be None.

deborahgu replied:
good point, change made.

deborahgu committed:
chore: improving the default case

code review suggestion was a good one: improve the default case, because
the fallback should be None.

justinhynes reviewed (approved):
Thank you for fighting the good fight.
